### PR TITLE
Update ngrest plugin date support date format.

### PR DESCRIPTION
**Pull request by vuongminh**
Date format in plugin date. User can customize date format

```
return [
'date' => ['date', 'dateFormat' => 'shortDate']
];
```

### CODE REVIEW

**Comment by nadar** [comment]
I have to test it, but looks good so far! Thanks. Could you please add in the `dateFormat` property phpdoc description add a list with common options the user can select? like

```php
@var string ....
+ `shortDate` Equivalent to 'M/d/yy' for en_US locale (e.g. 9/3/10) 
+ `dd.MM.yyyyy` Create custom patterns in order to .......
+ ...
...
@see https://docs.angularjs.org/api/ng/filter/date
```

So people can understand how to configure in order to support the local format.

**Comment by nadar** [comment]
(and of course it would be greate to see a unittest for the missing plugin, so at least we can be sure there is no php fatal error, even for future pull requests)

**Comment by nadar** [comment]
I assume your changes will not work as the date must be changed to an unix timestamp in order to send to the api. So there must be a rule in order to validate the provided timetsamp based on the current angular date input.

File: modules/admin/src/resources/js/directives.js [hunk 1]
```diff
@@ -659,6 +659,7 @@
             scope: {
                 "model": "=",
                 "options": "=",
+                "dateFormat": "=",
```
Comment (nadar) [inline modules/admin/src/resources/js/directives.js]: I assume this wont work:

> Note: These =attr attributes in the scope option of directives are normalized just like directive names. To bind to the attribute in <div bind-to-this="thing">, you'd specify a binding of =bindToThis.

From the angular docs (https://docs.angularjs.org/guide/directive)

Comment (nadar) [inline modules/admin/src/resources/js/directives.js]: I just fixed this problem, but it still won't work as of the validation. https://github.com/luyadev/luya/commit/0b5e79d811199372e8ca186a137ffe2f191eaa21

Comment (vuongminh) [inline modules/admin/src/resources/js/directives.js]: Sorry about my mistake! 

I will change validate

```
   			var res = n.split(".");
            			if (res.length == 3) {
            				if (res[2].length == 4) {
            					var en = res[1] + "/" + res[0] + "/" + res[2];
		        				$scope.model = (Date.parse(en)/1000);
		        				$scope.datePickerToggler = false;
            				}
}
```
to

```
$scope.model = (Date.parse(n)/1000);
$scope.datePickerToggler = false;
```

Are you think it ok?

**Comment by nadar** [comment]
I think this wont work either as other formats will not split according those rules (length then is not true for those false when using dd.mm.yyyy or yyyy.mm.dd oder mm.dd.yyyy: https://github.com/luyadev/luya/pull/1144/files#diff-710ff7c138d9bc6ec29088a032b77ddfR708

So i think we should find another solution:

- First of all i guess the date localisation should not be provided as plugin property, we should consider adding a new user database field so you can set your preffered localisation behavior in the user profile.
- The localisation must work then for date and datetime plugin.
- We have to validate the input date trough another mechanism instead of splitting strings, maybe using something like this: https://github.com/urish/angular-moment.
- The user defined localisation date should also be used by all phpsides strotime() uses cases.

What do you think about that?

So your pull request regarding localized time is very important and i am thankful for that!

**Comment by nadar** [comment]
Something else: You have changed the spacing and folding of variables of the complete directive.js files, have you used a fixer for that similar to the php-cs-fixer, so i could that over all files in order to ensure the same coding style. (But when just creating a pull request its not recommend, as we are unable to find the actual changes)

**Comment by vuongminh** [comment]
I think number 2 is good define date localistation and use in date and date time plugin!

And i am very sorry about my mistake because it is a first time i am make Prs :D! On next Pr i will carefull check before push!

**Pull request closed without merging**